lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -1,178 +1,164 @@
-          end
-        end
-
-        def relationship_value_for(serializer, options = {})
-          if serializer.respond_to?(:each)
-            serializer.map { |s| resource_identifier_for(s) }
-          else
-            if options[:virtual_value]
-              options[:virtual_value]
-            elsif serializer && serializer.object
-              resource_identifier_for(serializer)
-            end
-          end
-        end
-
-        def relationships_for(serializer)
-          Hash[serializer.associations.map { |association| [association.key, { data: relationship_value_for(association.serializer, association.options) }] }]
-        end
-
-        def included_for(serializer)
-          serializer.associations.flat_map { |assoc| _included_for(assoc.key, assoc.serializer) }.uniq
-        end
-
-        def _included_for(resource_name, serializer, parent = nil)
-          if serializer.respond_to?(:each)
-            serializer.flat_map { |s| _included_for(resource_name, s, parent) }.uniq
-          else
-            return [] unless serializer && serializer.object
-            result = []
-            resource_path = [parent, resource_name].compact.join('.')
-
-            if include_assoc?(resource_path)
-              primary_data = primary_data_for(serializer, @options)
-              relationships = relationships_for(serializer)
-              primary_data[:relationships] = relationships if relationships.any?
-              result.push(primary_data)
-            end
-
-            if include_nested_assoc?(resource_path)
-              non_empty_associations = serializer.associations.select(&:serializer)
-
-              non_empty_associations.each do |association|
-                result.concat(_included_for(association.key, association.serializer, resource_path))
-                result.uniq!
-              end
-            end
-            result
-          end
-        end
-
-        def include_assoc?(assoc)
-          check_assoc("#{assoc}$")
-        end
-
-        def include_nested_assoc?(assoc)
-          check_assoc("#{assoc}.")
-        end
-
-        def check_assoc(assoc)
-          @options[:include].any? { |s| s.match(/^#{assoc.gsub('.', '\.')}/) }
-        end
-
-        def add_links(options)
-          links = @hash.fetch(:links) { {} }
-          collection = serializer.object
-          @hash[:links] = add_pagination_links(links, collection, options) if paginated?(collection)
-        end
-
-        def add_pagination_links(links, resources, options)
-          pagination_links = ActiveModel::Serializer::Adapter::JsonApi::PaginationLinks.new(resources, options[:context]).serializable_hash(options)
-          links.update(pagination_links)
-        end
-
-        def paginated?(collection)
-          collection.respond_to?(:current_page) &&
-            collection.respond_to?(:total_pages) &&
-            collection.respond_to?(:size)
-        end
+  extend ActiveSupport::Autoload
```

Comment:
it's supposed to be so that the diffs are easy to read..  e.g. so we can https://github.com/rails-api/active_model_serializers/pull/1138